edge_utils.c

```diff
@@ -2378,6 +2380,14 @@ static int edge_init_routes(n2n_edge_t *eee, n2n_route_t *routes, uint16_t num_r
       custom_route.net_bitlen = 1;
       custom_route.gateway = route->gateway;
 
+      if(routectl(RTM_NEWROUTE, NLM_F_CREATE | NLM_F_EXCL, &custom_route, eee->device.if_idx) < 0)
+	return(-1);
+
+      /* ip route add 128.0.0.0/1 via n2n_gateway */
```

Comment:
Can you elaborate what's special about about 0.0.0.0/1 and 128.0.0.0/1?

Reply:
This only works in linux, it is explained in https://github.com/zerotier/ZeroTierOne/issues/178#issuecomment-204599227 . We use 0.0.0.0/1 and 128.0.0.0/1 (which covers the same IPv4 range as 0.0.0.0/0) to avoid touching the system default gateway (0.0.0.0/0). This has the advantage that if n2n is closed abnormally, the `0.0.0.0/1` and `128.0.0.0/1` routes are automatically deleted (since they are bound to the n2n interface) thus restoring the original routing table. In constrast, if we touch the default gateway and n2n is closed abnormally, the system would not have a default gateway route anymore

Reply:
Can you please put this explanation in a code comment, and as already discussed for other tickets, if this is a Linux-only change try to keep on a separate file ?